drivers/serial/uart_stm32.c

```diff
@@ -52,6 +55,10 @@ LOG_MODULE_REGISTER(uart_stm32);
 
 #define TIMEOUT 1000
 
+#ifdef CONFIG_PM_DEVICE
+char tmp;
```

Comment:
What's the point of this variable? Is there some weird reading optimization problem?

Reply:
Its scope could likely be at `while (LL_USART_IsActiveFlag_RXNE(UartInstance)) {` loop

Reply:
var is used to flush the rx fifo, now moved in the function itself